test/integ/test_multi_statement.py

```diff
@@ -0,0 +1,385 @@
+                time.sleep(1)
+            while con.is_still_running(con.get_query_status(sf_qid2)):
+                time.sleep(1)
+            cur.execute("drop table uselessTable")
+            assert cur.fetchall() == [("USELESSTABLE successfully dropped.",)]
+            cur.get_results_from_sfqid(sf_qid1)
+            assert cur.fetchall() == [(1,)]
+            assert cur.nextset() == cur
+            assert cur.fetchall() == [("a",)]
+            assert cur.nextset() == cur
+            assert cur.fetchall() == [("row1", 1), ("row2", 2), ("row3", 3)]
+            assert cur.nextset() is None
+            assert not cur.fetchall()
+            cur.get_results_from_sfqid(sf_qid2)
+            assert cur.fetchall() == [(2,)]
+            assert cur.nextset() == cur
+            assert cur.fetchall() == [("b",)]
+            assert cur.nextset() == cur
+            assert cur.fetchall() == [("row1", 1), ("row2", 2), ("row3", 3)]
+            assert cur.nextset() is None
+            assert not cur.fetchall()
+
+
+def test_done_caching_multi(conn_cnx):
+    """Tests whether get status caching is working as expected."""
+    with conn_cnx(session_parameters={PARAMETER_MULTI_STATEMENT_COUNT: 0}) as con:
+        with con.cursor() as cur:
+            cur.execute_async(
+                "select 1; select 'a'; select count(*) from table(generator(timeLimit => 2));"
+            )
+            qid1 = cur.sfqid
+            cur.execute_async(
+                "select 2; select 'b'; select count(*) from table(generator(timeLimit => 2));"
+            )
+            qid2 = cur.sfqid
+            assert len(con._async_sfqids) == 2
+            time.sleep(5)
+            while con.is_still_running(con.get_query_status(qid1)):
+                time.sleep(1)
+            assert con.get_query_status(qid1) == QueryStatus.SUCCESS
+            assert len(con._async_sfqids) == 1
+            assert len(con._done_async_sfqids) == 1
+            time.sleep(5)
+            while con.is_still_running(con.get_query_status(qid2)):
+                time.sleep(1)
+            assert con.get_query_status(qid2) == QueryStatus.SUCCESS
+            assert len(con._async_sfqids) == 0
+            assert len(con._done_async_sfqids) == 2
+            assert con._all_async_queries_finished()
+
+
+def test_desc_rewrite_multi(conn_cnx, caplog):
+    """Tests whether multiple describe queries are rewritten properly and that this is logged."""
+    with conn_cnx(session_parameters={PARAMETER_MULTI_STATEMENT_COUNT: 0}) as con:
+        with con.cursor() as cur:
+            table1 = random_string(5, "test_desc_rewrite_multi_")
+            table2 = random_string(5, "test_desc_rewrite_multi_")
+            cur.execute(
+                f"create temp table {table1} (a int); create temp table {table2} (b int);"
+            )
+            caplog.set_level(logging.DEBUG, "snowflake.connector")
+            cur.execute(f"desc {table1}; select 3; desc {table2}; select 'b';")
```

Comment:
Once we update the regex I'd like to see `[ TYPE =  { COLUMNS | STAGE } ]` being tested too from https://docs.snowflake.com/en/sql-reference/sql/desc-table.html